freshmaker/models.py

```diff
@@ -626,6 +626,23 @@ def get_lowest_build_id(cls, session):
             return 0
         return build.build_id
 
+    @classmethod
+    def get_most_original_nvr(cls, nvr):
+        """
+        Get original NVR recursively until reach the one which was not built by freshmaker
+
+        Return the NVR of most original image
+        """
+        original_nvr = None
+        build = db.session.query(ArtifactBuild).filter(cls.rebuilt_nvr == nvr).first()
+        while build:
+            original_nvr = build.original_nvr
+            build = db.session.query(ArtifactBuild).filter(cls.rebuilt_nvr == original_nvr).first()
+        # This can be called for many times while resolving images, close the session
+        # to avoid possible flask-SqlAlchemy QueuePool limit size overflow
+        db.session.close()
```

Comment:
I am not, but is it safe to close the session here? If it's used further, will it be opened again automatically or how this works?

Reply:
SQLAlchemy session is a factory, a new session will be created automatically when the old one is closed. But yes, there is a risk if someone call this API after some instance is attached to the session but before commit, this can result in issues, so probably it's more safe to increase the SQLAlchemy pool size or even disabling pooling.
So I've removed this and opened an internal MR to change the pool size for our deployment. I didn't add that in default config because sqlite is used by default, which is using `NullPool`, it doesn't have pooling and doesn't accept `pool_size` as engine option.